[v4] Refactor Proxy documentation (#475)
strapi/documentation: docs/developer-docs/latest/setup-deployment-guides/deployment/optional-software/nginx-proxy.md

The below configuration is based on Nginx virtual hosts, this means that you create configurations for each **domain** to allow serving multiple domains on the same port such as 80 (HTTP) or 443 (HTTPS). It also uses a central upstream file to store an alias to allow for easier management, load balancing, and failover in the case of clustering multiple Strapi deployments.

!!!include(developer-docs/latest/setup-deployment-guides/deployment/optional-software/snippets/strapi-server.md)!!!

### Nginx Upstream

> Upstream blocks are used to map an alias such as `strapi` to a specific URL such as `localhost:1337`. While it would be useful to define these in each virtual host file, Nginx currently doesn't support loading these within the virtual host **if you have multiple virtual host files** and instead you should configure these within the `conf.d` directory as this is loaded before any virtual host files.

Reviewer: ```suggestion
Upstream blocks are used to map an alias such as `strapi` to a specific URL such as `localhost:1337`. While it would be useful to define these in each virtual host file, Nginx currently doesn't support loading these within the virtual host if you have multiple virtual host files. Instead, configure these within the `conf.d` directory as this is loaded before any virtual host files.
```

---

> - Example Domain: `api.example.com`

Reviewer: ```suggestion
- Example domain: `api.example.com`
```

> - Example Admin: `api.example.com/admin`

Reviewer: ```suggestion
- Example admin panel: `api.example.com/admin`
```

- Example API: `api.example.com/api`

> - Example Uploaded Files (local provider): `api.example.com/uploads`

Reviewer: ```suggestion
- Example uploaded Files (local provider): `api.example.com/uploads`
```

> #### Sub-Folder Unified

Reviewer: ```suggestion
#### Sub-Folder unified
```

> This config is using a sub-folder that is dedicated to Strapi only. It will redirect normal HTTP traffic over to SSL and hosts the "frontend" files on `/var/www/html` like a normal web server, but proxies all strapi requests on the `example.com/test` sub-path.

Reviewer: ```suggestion
This configuration is using a sub-folder that is dedicated to Strapi only. It will redirect normal HTTP traffic over to SSL and hosts the front-end files on `/var/www/html` like a normal web server, but proxies all strapi requests on the `example.com/test` sub-path.
```
